core/src/main/java/org/testcontainers/containers/ComposeConfiguration.java

```diff
@@ -0,0 +1,382 @@
+            followLogs(containerId, new Slf4jLogConsumer(log).withPrefix(container.getNames()[0]));
+        }
+        //follow logs using registered consumers for this service
+        logConsumers
+            .getOrDefault(serviceName, Collections.emptyList())
+            .forEach(consumer -> followLogs(containerId, consumer));
+        serviceInstanceMap.putIfAbsent(serviceName, containerInstance);
+    }
+
+    private void waitUntilServiceStarted(String serviceName, ComposeServiceWaitStrategyTarget serviceInstance) {
+        final WaitAllStrategy waitAllStrategy = waitStrategyMap.get(serviceName);
+        if (waitAllStrategy != null) {
+            waitAllStrategy.waitUntilReady(serviceInstance);
+        }
+    }
+
+    private String getServiceNameFromContainer(com.github.dockerjava.api.model.Container container) {
+        final String containerName = container.getLabels().get("com.docker.compose.service");
+        final String containerNumber = container.getLabels().get("com.docker.compose.container-number");
+        return String.format("%s_%s", containerName, containerNumber);
+    }
+
+    public void runWithCompose(boolean localCompose, String cmd) {
+        runWithCompose(localCompose, cmd, Collections.emptyMap());
+    }
+
+    public void runWithCompose(boolean localCompose, String cmd, Map<String, String> env) {
+        Preconditions.checkNotNull(composeFiles);
+        Preconditions.checkArgument(!composeFiles.isEmpty(), "No docker compose file have been provided");
+
+        final DockerCompose dockerCompose;
+        if (localCompose) {
+            dockerCompose = new LocalDockerCompose(this.executable, composeFiles, project);
+        } else {
+            dockerCompose = new ContainerisedDockerCompose(this.defaultImageName, composeFiles, project);
+        }
+
+        dockerCompose.withCommand(cmd).withEnv(env).invoke();
+    }
+
+    void registerContainersForShutdown() {
+        ResourceReaper
+            .instance()
+            .registerLabelsFilterForCleanup(Collections.singletonMap("com.docker.compose.project", project));
+    }
+
+    @VisibleForTesting
+    List<Container> listChildContainers() {
+        return dockerClient
+            .listContainersCmd()
+            .withShowAll(true)
+            .exec()
+            .stream()
+            .filter(container -> Arrays.stream(container.getNames()).anyMatch(name -> name.startsWith("/" + project)))
+            .collect(Collectors.toList());
+    }
+
+    void startAmbassadorContainer() {
+        if (!this.ambassadorPortMappings.isEmpty()) {
+            this.ambassadorContainer.start();
+        }
+    }
+
+    public void withExposedService(String serviceName, int servicePort) {
+        withExposedService(serviceName, servicePort, Wait.defaultWaitStrategy());
+    }
+
+    public void withExposedService(String serviceName, int instance, int servicePort) {
+        withExposedService(serviceName + "_" + instance, servicePort);
```

Comment:
Don't we need to be version aware here? There are multiple instances of the seperator here in this class.

Reply:
yes, I miss it in this version but as we discussed internally will be adding the change soon.